dart/lib/src/protocol/event.dart

```diff
@@ -9,6 +9,7 @@ import '../version.dart';
 class Event {
   /// Creates an event.
   const Event({
+    this.eventId,
```

Comment:
@bruno-garcia @rxlabz the problem here is:

I can't do:
`this.eventId = SentryId.newId(),`
a optional default param must be a constant.

also I cannot do `final SentryId eventId = SentryId.newId();`
or the class cant be immutable (const ctor) as it is inited with a non constant value.

ideas?

Reply:
I think you can remove the `const` ctor. All the fields are final + `@immutable` are enough. Dart `const` are particular.

> "const" has a meaning that's a bit more complex and subtle in Dart.
from https://news.dartlang.org/2012/06/const-static-final-oh-my.html

Reply:
not really, I've tried already:

the default value of an optional parameter must be constant.dartnon_constant_default_value

Reply:
I found this:

```
class ClassB {
  final ClassA classA;
  ClassB({this.classA = const ClassA()}); // <- creation is const
}
```

but it didnt work with SentryId at least

Reply:
this should do the job :
```dart
Event({
    SentryId eventId ,
    this.loggerName,
    this.serverName,
    this.release,
    this.environment,
    this.message,
    this.transaction,
    this.exception,
    this.stackTrace,
    this.level,
    this.culprit,
    this.tags,
    this.extra,
    this.fingerprint,
    this.userContext,
    this.contexts,
    this.breadcrumbs,
    this.sdk,
  }):eventId = SentryId.newId();
```

Reply:
if I do:

`this.eventId = const SentryId.internal(''),`

`const SentryId.internal(this._id);` it'd work but then we'd run the same problem on SentryId class

Reply:
this should work
```dart
Event({
    SentryId eventId,
    this.loggerName,
    this.serverName,
    this.release,
    this.environment,
    this.message,
    this.transaction,
    this.exception,
    this.stackTrace,
    this.level,
    this.culprit,
    this.tags,
    this.extra,
    this.fingerprint,
    this.userContext,
    this.contexts,
    this.breadcrumbs,
    this.sdk,
  }) : eventId = eventId ?? SentryId.newId();
```

Reply:
strangely the analyzer prefers this to `: this.eventId = eventId ?? SentryId.newId();` 🤷‍♂️

Reply:
mmmm I didn't know about the mix of syntactic sugar + named params, nice

Reply:
Would it work to drop `this`, declare the field, add a body to the ctor and add?

`this.eventId = eventId ?? SentryId();`

Reply:
solved doing this: https://github.com/getsentry/sentry-dart/pull/119#discussion_r508693328

thank you both